src/Asset.sol

```diff
@@ -0,0 +1,199 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+
+import '@openzeppelin/contracts/token/ERC20/IERC20.sol';
+import '@openzeppelin/contracts/token/ERC20/ERC20.sol';
+import '@openzeppelin/contracts/token/ERC20/extensions/ERC20Capped.sol';
+import '@openzeppelin/contracts/access/Ownable.sol';
+import '@openzeppelin/contracts/security/Pausable.sol';
+
+/// Lets you purchase new tokens, and then redeem them.
+contract Vendor is Ownable {
+    uint16 private _feeInBasisPoints;
+
+    event Purchased(
+        address indexed asset,
+        address indexed user,
+        uint256 amount
+    );
+    event Redeemed(
+        address indexed asset,
+        address indexed user,
+        uint256 amount,
+        string uri
+    );
+    event SetFee(uint16 feeInBasisPoints);
+    event SetOwner(address owner);
+
+    constructor() {
+        _feeInBasisPoints = 100;
+        _transferOwnership(msg.sender);
```

Comment:
I could test this, but I think when a contract `is Ownable`, the ownership is transfered to the deployer by default: https://github.com/OpenZeppelin/openzeppelin-contracts/blob/8b778fa20d6d76340c5fac1ed66c80273f05b95a/contracts/access/Ownable.sol#L13